### PR TITLE
add support and test for NIP-11 Relay Information Document

### PR DESCRIPTION
**Pull request by bryanmontz**
This PR adds support for NIP-11 Relay Information Document decoding. These properties will give clients all the necessary information a relay offers about its service.

### CODE REVIEW

**Comment by bryanmontz** [comment]
Also disabled the "nesting" SwiftLint rule that says that types may only be nested to one level. With the json objects in the Nostr protocol spec, deeper nesting makes sense.

File: Sources/NostrSDK/RelayInfo.swift [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+// NIP-11 Relay Information Document
+// https://github.com/nostr-protocol/nips/blob/master/11.md
+
+public struct RelayInfo: Codable {
+    public let name: String?
+    public let description: String?
+    public let contactPubkey: String?
+    public let alternativeContact: String?
+    public let supportedNIPs: [Int]?
+    public let supportedNIPExtensions: [String]?
```
Comment (tyiu) [inline Sources/NostrSDK/RelayInfo.swift:19]: Is this to spec? I don't see it.

File: Tests/NostrSDKTests/Fixtures/relay_info.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        "auth_required": false,
+        "payment_required": true
+    },
+    "retention": [
+        { "kinds": [0, 1, [5, 7], [40, 49]], "time": 3600 },
+        { "kinds": [[40000, 49999]], "time": 100 },
+        { "kinds": [[30000, 39999]], "count": 1000 },
+        { "time": 3600, "count": 10000 }
+    ],
+    "language_tags": [ "en", "en-419" ],
```
Comment (tyiu) [inline Tests/NostrSDKTests/Fixtures/relay_info.json:36]: I think the example in the spec is slightly wrong. en-419 would mean English spoken in Latin America. They probably mean es-419, which is Spanish spoken in Latin America. But no big deal.

File: Tests/NostrSDKTests/Fixtures/relay_info.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    "pubkey": "test-pubkey",
+    "contact": "mailto:somebody@nostr.test",
+    "supported_nips": [
+        1,
+        2,
+        4,
+        9
+    ],
+    "supported_nip_extensions": [
+        "11a"
```
Comment (tyiu) [inline Tests/NostrSDKTests/Fixtures/relay_info.json:13]: Not sure what this is.

**Review by tyiu — APPROVED** [review]

**Pull request merged**